src/app/workspace/resource/resource-instance-form/resource-instance-form.component.ts

```diff
@@ -96,96 +69,102 @@ export class ResourceInstanceFormComponent implements OnInit, OnChanges, OnDestr
 
-        // use response to go further with properties
-        this.updateParent.emit({ title: this.resourceLabel, subtitle: 'Set the property values of the resource' });
-    }
+        this.preparing = true;
+        this.loading = true;
+        this._dspApiConnection.v2.ontologyCache.reloadCachedItem(this.ontologyIri).subscribe(
+            (res: ReadOntology) => {
+                this._dspApiConnection.v2.ontologyCache.getResourceClassDefinition(resourceClassIri).subscribe(
+                    (onto: ResourceClassAndPropertyDefinitions) => {
+                        this.ontologyInfo = onto;
+
+                        this.resourceClass = onto.classes[resourceClassIri];
+
+                        // set label from resource class
+                        const defaultClassLabel = this.defaultClasses.find(i => i.iri === this.resourceClass.subClassOf[0]);
+                        this.resourceLabel = this.resourceClass.label + (defaultClassLabel ? ' (' + defaultClassLabel.label + ')' : '');
+
+                        // filter out all props that cannot be edited or are link props but also the hasFileValue props
+                        this.properties = onto.getPropertyDefinitionsByType(ResourcePropertyDefinition).filter(
+                            prop =>
+                                !prop.isLinkProperty &&
+                                prop.isEditable &&
+                                prop.id !== Constants.HasStillImageFileValue &&
+                                prop.id !== Constants.HasDocumentFileValue &&
+                                prop.id !== Constants.HasAudioFileValue &&
+                                prop.id !== Constants.HasMovingImageFileValue &&
+                                prop.id !== Constants.HasArchiveFileValue &&
+                                prop.id !== Constants.HasTextFileValue
+                                // --> TODO for UPLOAD: expand with other representation file values
+                        );
+
+                        if (onto.properties[Constants.HasStillImageFileValue]) {
+                            this.hasFileValue = 'stillImage';
+                        } else if (onto.properties[Constants.HasDocumentFileValue]) {
+                            this.hasFileValue = 'document';
+                        } else if (onto.properties[Constants.HasAudioFileValue]) {
+                            this.hasFileValue = 'audio';
+                        } else if (onto.properties[Constants.HasMovingImageFileValue]) {
+                            this.hasFileValue = 'movingImage';
+                        } else if (onto.properties[Constants.HasArchiveFileValue]) {
+                            this.hasFileValue = 'archive';
+                        } else if (onto.properties[Constants.HasTextFileValue]) {
+                            this.hasFileValue = 'text';
+                        } else {
+                            this.hasFileValue = undefined;
+                        }
```

Comment:
You could use switch case in that case

Reply:
we have this code in a few places in the app so if we want to change it to a switch case, we should change it everywhere but then that kind of exceeds the goal of this PR